src/hotspot/share/nmt/vmatree.cpp

```diff
@@ -86,6 +86,19 @@ VMATree::SummaryDiff VMATree::register_mapping(position A, position B, StateType
 
     // Direct address match.
     if (leqA_n->key() == A) {
+      // For NMT reports in detail mode, separate stacks are required for Reserve and Commit operations.
+      // So, do not touch the stack of the node since it is for Reserve operation. Instead store the new
+      // call-stack in the second_stack of the node.
+      if ((leqA_n->val().out.mem_tag() != leqA_n->val().in.mem_tag()
+           || leqA_n->val().out.type() != leqA_n->val().in.type())          // leqA_n is the start of a reserved region
+          && !(state == StateType::Reserved && !use_tag_inplace)            // we are not reserving a new region
+          && !NativeCallStackStorage::is_invalid(leqA_n->val().out.stack()) // the primary stack is already filled
+         ) {
+        stA.out.set_stack(leqA_n->val().out.stack());
+        stB.in.set_stack(leqA_n->val().out.stack());
+        stA.out.set_second_stack(metadata.stack_idx);
+        stB.in.set_second_stack(metadata.stack_idx);
```

Comment:
Did we consider naming them differently?

One obvious choice would be ReserveStack and CommitStack?

Right now, we need to mentally map them one way, and I am afraid that we could possibly get it reversed at some other point in code...

Reply:
One StackIndex is used for all reserve/commit/uncommit/release operations.
We need another one just used for committing and uncommitting the start-node of the reserved region.
So names like ReserveXXX and CommitXXX would not fit. 
I do not insist on the chosen names and am open to proper alternatives.

Reply:
In your gtest you have a test called "SeparateStacksForCommitAndReserve", which seems to say to me, that commit and reserve would be good names?

Reply:
The full name should be something like:
commit_stack_at_start_of_region
or
stack_for_commit_at_region_base
Although it is also used for uncommit but we don't show uncommit stack traces in NMT report. 
What do you think? Any other better naming?